rapid_response_xblock/static/js/src/rapid.js

```diff
@@ -291,25 +307,49 @@
     }
 
     /**
-     * Render the chart in the container.
+     * Get a selected run id, or undefined if there are no runs
      *
-     * @param {Object} container D3 selector for the chart container
-     * @param {number} chartIndex The index of the chart (either 0 or 1)
+     * @param {number} chartIndex Index of the chart
+     * @returns {number} A run id
      */
-    function renderChart(container, chartIndex) {
+    function getSelectedRun(chartIndex) {
       var runs = state.runs;
-      var counts = state.counts;
-      var choices = state.choices;
       var selectedRun = state.selectedRuns[chartIndex];
 
-      // select the proper option and use it to filter the runs
-      var select = container.select(".selection-container").select("select")
-        .classed("hidden", state.runs.length === 0 || state.is_open);
       if (selectedRun === null && runs.length > 0) {
         // The newest run should be the most recent one according to the info received from the server.
         selectedRun = runs[0].id;
       }
 
+      return selectedRun;
+    }
+
+    /**
+     * Count number of responses for a run
+     * @param {number} runId The run id
+     *
+     * @returns {number} The total number of responses
+     */
+    function numResponses(runId) {
+      var total = 0;
+      state.choices.forEach(function(item) {
+        total += state.counts[item.answer_id][runId] || 0;
+      });
+      return total;
```

Comment:
j/w: Is this logic is going to change when the API returns the total counts (#57)?

Reply:
Yes it should